docs/upgrading.md

```diff
@@ -2,6 +2,98 @@
+
+### Beta release
+
+Contour 0.15.0 remains the current stable release.
+The `:latest` tag will continue to point to 0.15.0 until Contour 1.0.0 is released.
+
+### Deprecated deployments
+
+The following deployment examples were deprecated in Contour 0.15 and have been removed:
+
+- `deployment-grpc-v2`
+- `ds-grpc-v2`
+- `ds-hostnet-split`
+- `ds-hostnet`
+
+### IngressRoute v1beta1 deprecation
+
+The IngressRoute v1beta1 CRD has been deprecated and will not receive further updates.
+Contour will continue to recognise IngressRoute v1beta1 through Contour 1.0.0 final, however we anticipate it will be removed completely shortly after that.
+
+The replacement for IngressRoute which we have called HTTPProxy is available in Contour 1.0.0-beta.1 and is anticipated to be declared final by Contour 1.0.0.
+
+**TODO(dfc) link to HTTPProxy documentation**
+
+## The easy way to upgrade
+
+If the following are true for you:
+
+ * Your previous installation is in the `heptio-contour` namespace.
+ * You are using one of the [example](/example/) deployments.
+ * Your cluster can take few minutes of downtime.
+
+Then the simplest way to upgrade to 1.0.0-beta.1 is to delete the `heptio-contour` namespace and reapply the `examples/contour` sample manifest.
+From the root directory of the repository:
+```
+kubectl delete namespace heptio-contour
+
+kubectl apply -f examples/contour
+```
+Note that `examples/contour` now deploys into the `projectcontour` namespace.
+
+If you're using a `LoadBalancer` Service, deleting and recreating may change the public IP assigned by your cloud provider.
+You'll need to re-check where your DNS names are pointing as well, using [Get your hostname or IP address](./deploy-options.md#get_your_hostname_or_ip_address).
+
+## The less easy way
+
+This section contains information for administrators who wish to apply the Contour 0.15 to 1.0.0-beta.1 changes manually.
+
+### Namespace change
+
+As part of sunsetting the Heptio brand the `heptio-contour` namespace has been renamed to `projectcontour`.
+Contour assumes it will be deployed into the `projectcontour` namespace.
+
+If you deploy Contour into another namespace you will need to pass `contour bootstrap --namespace=<namespace>` and update the `contour.yaml` configuration file's leader election parameters as appropriate.
+
+### Upgrade to Contour 1.0.0-beta.1
+
+As part of sunsetting the Heptio brand Docker images have moved from `gcr.io/heptio-images` to `docker.io/projectcontour`.
+
+Change the Contour image version to `docker.io/projectcontour/contour:v1.0.0-beta.1`.
+
+### Recommended Envoy version
+
+The recommended version of Envoy remains unchanged from Contour 0.15.
+Ensure the Envoy image version is `docker.io/envoyproxy/envoy:v1.11.1`.
+
+### Leader Election
+
+Contour 1.0.0-beta.1 enables leader election by default.
+No specific configuration is required 
```

Comment:
No specific configuration is needed for this feature, but it will use a ConfigMap named `leader-elect` in the `projectcontour` namespace. These details can be changed using the config file.

(or something).

Reply:
Also, we need to note that leader election now does *not* block the gRPC endpoint from becoming available, that is, all Contours now serve xDS.

Reply:
@youngnick do you want to take a stab at this; you can push to this branch.

Reply:
Or we merge as is and you send a followup PR. :shrug:

Reply:
I'll do a followup PR.